### PR TITLE
ARROW-12773: [Docs] Clarify Java support for ORC and Parquet via JNI bindings

### PR DESCRIPTION
**Pull request by hcoona**
See https://issues.apache.org/jira/browse/ARROW-12773 for further details.

### CODE REVIEW

**Comment by lidavidm** [comment]
Good catch on the ORC adapter! However, I think Parquet is supported, just in a different way: there's a separate artifact that provides JNI bindings to the Arrow Datasets library, which in turn reads Parquet: https://github.com/apache/arrow/tree/master/java/dataset

It might be good to clarify that in the docs, though (e.g. by linking to or naming the actual Java artifacts that provide the support).

**Comment by hcoona** [comment]
Nice to know it. The MR updated for it.

File: docs/source/status.rst [hunk 1]
```diff
@@ -232,6 +232,8 @@ Notes:
 
 * *W* = Write supported
 
-* \(1) Nested read/write not supported
+* \(1) Nested read/write not supported.
 
-* \(2) Through JNI bindings to datasets.
+* \(2) Through JNI bindings. (Provided by ```org.apache.arrow.orc:arrow-orc```)
+
+* \(3) Through JNI bindings to datasets. (Provided by ```org.apache.arrow:arrow-dataset```)
```
Comment (lidavidm) [inline docs/source/status.rst:239]:
```suggestion
* \(2) Through JNI bindings. (Provided by ``org.apache.arrow.orc:arrow-orc``)

* \(3) Through JNI bindings to Arrow C++ Datasets. (Provided by ``org.apache.arrow:arrow-dataset``)
```

**Review by lidavidm — APPROVED** [review]
Thanks! Just one small syntax nit (reST uses double-backticks for monospace text)

**Review by lidavidm — APPROVED** [review]

**Comment by lidavidm** [comment]
Thanks for the fix!

**Pull request closed without merging**